Lib/test/test_subprocess.py

```diff
@@ -2018,11 +2018,10 @@ def test_user(self):
                                 user=user,
                                 close_fds=close_fds)
                     except PermissionError as e:  # (EACCES, EPERM)
-                        self.assertIsNone(e.filename)
-                    except OSError as e:
-                        if e.errno not in (errno.EACCES, errno.EPERM):
-                            raise
-                        self.assertIsNone(e.filename)
+                        if e.errno == errno.EACCES:
+                            self.assertEqual(e.filename, sys.executable)
+                        else:
+                            self.assertIsNone(e.filename)
```

Comment:
Strictly speaking, EPERM can be set in execve() in some circumstances. `e.filename` will be equal to `sys.executable` then. But I do not think that they are related to Python tests.